managed/services/versioncache/versioncache.go

```diff
@@ -133,52 +141,35 @@ func (s *Service) findServiceForUpdate() (*service, error) {
 	return results, nil
 }
 
-func softwareName(s agents.Software) (models.SoftwareName, error) {
-	var softwareName models.SoftwareName
-	switch software := s.(type) {
-	case *agents.Mysqld:
-		softwareName = models.MysqldSoftwareName
-	case *agents.Xtrabackup:
-		softwareName = models.XtrabackupSoftwareName
-	case *agents.Xbcloud:
-		softwareName = models.XbcloudSoftwareName
-	case *agents.Qpress:
-		softwareName = models.QpressSoftwareName
-	default:
-		return "", errors.Errorf("invalid software type %T", software)
-	}
-
-	return softwareName, nil
-}
-
 // updateVersionsForNextService tries to find a service that needs update and performs update if such service is found.
 // Returns desired time interval to wait before next call of the updateVersionsForNextService.
 func (s *Service) updateVersionsForNextService() (time.Duration, error) {
-	foundService, err := s.findServiceForUpdate()
+	serviceForUpdate, err := s.findServiceForUpdate()
 	if err != nil {
 		return minCheckInterval, err
 	}
 
-	if !foundService.WaitNextCheck {
-		return foundService.CheckAfter, nil
+	if !serviceForUpdate.WaitNextCheck {
+		return serviceForUpdate.CheckAfter, nil
+	}
+
+	softwareList := serviceForUpdate.BackupSoftwareList
+	if len(softwareList) == 0 {
```

Comment:
is there any reason we can't just use `len(serviceForUpdate.BackupSoftwareList)` instead of the extra variable?

It seems to be the only place we use this field and currently we just re-assign it

Reply:
We can remove the alias, but in that case we'll get longer code)